src/main/java/com/csoft/MainMojo.java

```diff
@@ -43,7 +43,17 @@
     @Parameter(property = "failBuildOnBlacklisted", defaultValue = "false")
     private boolean failBuildOnBlacklisted;
 
-    private Log log = getLog();
+    public MainMojo(){
```

Comment:
do we need the public default constructor?

Reply:
You need it for maven itself, the other constructor is for the tests. If you prefer, I could replace the bigger constructor and replace it with setters for the test-doubles